File: seed/ruby-sdk-v2/server-sent-event-examples/lib/seed/completions/types/event_event.rb
# frozen_string_literal: true

module Seed
  module Completions
    module Types
      class EventEvent < Internal::Types::Model
        field :event, -> { String }, optional: false, nullable: false
        field :name, -> { String }, optional: false, nullable: false
      end
    end
  end
end

File: seed/php-model/server-sent-event-examples/src/Completions/EventEvent.php
<?php

namespace Seed\Completions;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class EventEvent extends JsonSerializableType
{
    /**
     * @var string $event
     */
    #[JsonProperty('event')]
    public string $event;

    /**
     * @var string $name
     */
    #[JsonProperty('name')]
    public string $name;

    /**
     * @param array{
     *   event: string,
     *   name: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->event = $values['event'];
        $this->name = $values['name'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
